Christina Kopecky - front-end-project

ckopecky opened the pull request:
## Trello Set Up:

* [x] Create a Trello account
* [x] Create a new board called "Lambda Notes - {Your Name}"
* [x] Create lists titled `backlog`,`To Do`, `In Progress`, and `Done`
* [x] Fill in the `To Do` list with the MVP features listed below
* [x] Fill in the `backlog` list with all the extra features listed below
* [x] Share your board with the project manager that has been assigned to you. If you have not been assigned yet, reach out to your lead TA for guidance
* [x] Add your Trello URL to your project's README.md file. Commit the change, push it to your repository & submit a pull request

## MVP Features:

* [x] Display a list of notes
* [ ] Create a note with a title and content
* [x] View an existing note
* [ ] Edit an existing note
* [ ] Delete an existing note

ckopecky committed:
added Trello URL to top of readme page

ckopecky committed:
created react-app; added react router

added node_modules to gitignore

ckopecky committed:
added component files

ckopecky committed:
added reactstrap to project

ckopecky committed:
rough sketch of sidebar completed

ckopecky committed:
card container added to sidebar - list component created

ckopecky committed:
Rough sketch of detailed note view

no route linkeage yet to edit or delete

ckopecky committed:
Removed the copies of the card to start thinking about making the component

reusable

ckopecky committed:
generated some dummydata so I can make cards a reusable component

modified List component

ckopecky committed:
routing between note List.js, Noteview.js and EditNote.js WORKS!

Nothing dynamic yet. Hard coding things until I get

basics on page

ckopecky committed:
have a delete modal up and running - no absolute functionality yet

just bare bones

ckopecky committed:
create note component done -

Bare bones in place! Router works!

Now to refactor..

ckopecky committed:
refactored react-router to its own file,

refactored Cards to be their own file

cleaned up code thus far to prepare for making it functional

ckopecky committed:
passed props through to cards and mapped over dummydata to create

100 unique notes

ckopecky committed:
changed favicon to a quill

ckopecky committed:
made link dynamic - now just have to figure out how to make content

pass down through props

ckopecky committed:
pushing for Dre'Sean

ckopecky committed:
push for dre-sean two

ckopecky committed:
refactored into redux

ckopecky committed:
added horizontal line in card

ckopecky committed:
installed react-router for some reason it wasn't in my dependencies

ckopecky committed:
deleted stuff so that I could start from scratch

implented styling on Sidebar

ckopecky committed:
basic nav bar, main container, and

card placeholder set

ckopecky committed:
notes mapped over

a bit of restructuring as far as the files go.

rough but it's presentational

ckopecky committed:
imported Router - not functional yet, but linked up to home page

ckopecky committed:
made static link out of title,

fixed background

fixed styling of navbar

fixed overall styline

figured out bug with router

--which was I needed to use EXACT path in order to get it to render to

page I wanted it to.

ckopecky committed:
figured out how to change link color

overflow changed to hidden

ckopecky committed:
finished Single Note View

ckopecky committed:
all stylings done except toggle on delete modal

ckopecky committed:
we have a modal that pops up with everything inside!! WOOT!

Now to actually delete something....

ckopecky committed:
changed buttons to divs on SingleNoteView

ckopecky committed:
saving for Moises - can't figure out what I'm doing wrong in CWM in

SingleNoteView

ckopecky committed:
Change placeholder txt to something from state to get feel for data flow

ckopecky committed:
routing to editNote works! WOOT!

ckopecky committed:
after lunch push - updating git

ckopecky committed:
updated stylings - managed to name some of the classNames

the same thing so some of the stylings mismatched

or didn't come across like they were supposed to

Looks much better now

ckopecky committed:
route to edit note is dynamic

ckopecky committed:
singleNoteView intact!! WOOT!!!!!!!!!

This has always been the hardest thing for me to tackle.

ckopecky committed:
MVP met -

have an interesting bug where if I go to try to delete something -

it acts as if it deletes it. Goes back to home page. . .

. . . then when I click on another note, it will return an error

ckopecky committed:
some tweaks and updates including:

shortened dummydata to handle not so much data

added values to the edit form so content shows up in form

updated state in notecards so that the reversed array will show up CWM

we have a front-end that looks pretty - just not much function.

designerexpert commented:
# Comments
* I was very impressed with your project during MBP week. You achieved a finished project, and it looks like you referenced some of the walk-through code, but implemented it creatively on your own version.
* I can see that you have a much stronger grasp on the concepts of React, Props. And with more repetitions you will do great.
* I suggest maintaining this project or another of your choosing to keep practicing react + redux skills as you continue forward.
* Keep up the commit messages and the often commits, they tell a great story about the project, and help a bunch when working in teams 👍

ckopecky committed:
updated package.json file
created server.js file
added GET, POST, PUT and DELETE requests for notes

ckopecky committed:
moved files to a client folder to separate front end and back end

create package-json file for client to show dependencies for front-end

ckopecky committed:
removed client will redo client-react-app

ckopecky committed:
notecards mapped and singlenoteview established

was able to pass state through the Link in the "to" attribute as an obj

some styling done

a couple of wind, so am committing

ckopecky committed:
added edit and delete divs

linked to editNote component

ckopecky committed:
edit note form and create note form created

sidebar routes to home page and to create page

ckopecky committed:
fixed styling

ckopecky committed:
delete modal created and styled

ckopecky committed:
edit form has values that call up the current note's contents

ckopecky committed:
Move files around so that server is in root of directory

ckopecky committed:
Add .env file to gitignore

ckopecky committed:
Add backend dependencies

ckopecky committed:
Server with a basic sanity check with 100 dummy notes

ckopecky committed:
Back endpoints written - checked via postman

ckopecky committed:
Schema written

ckopecky committed:
Add endpoint to get fakerdata

ckopecky committed:
Modify sanity check to be simple

ckopecky committed:
Add todo for corsoptions

ckopecky committed:
Update frontend packages

ckopecky committed:
Update frontend packages

ckopecky committed:
Move Sidebar to outside router switch

ckopecky committed:
Add fakerdata function

ckopecky committed:
Add a CDM to put data on state

ckopecky committed:
Allows sidebar and rest of page to be side by side

ckopecky committed:
get rid of dummydata and refactor fakerdata.js

ckopecky committed:
Add constructor and state to App

ckopecky committed:
Add CDM and handleChange methods to App.js

ckopecky committed:
render props through routes

ckopecky committed:
change css file to App.css to break up code

import axios

ckopecky committed:
Add App.css to break up css code

ckopecky committed:
Add CreateNote.css to break up code

ckopecky committed:
Put header in its own file

ckopecky committed:
Change header path

ckopecky committed:
Add notes prop

ckopecky committed:
move CDM to App.js

ckopecky committed:
Add faker uuid

ckopecky committed:
SingleNoteView Corrections

ckopecky committed:
Corrections

ckopecky committed:
Update body of form to textarea

update values to the props that are passed from the editnote.js

ckopecky committed:
pass props to form from this.props.location.state

ckopecky committed:
pass props to form

ckopecky committed:
Take out link to google fonts

ckopecky committed:
Delete import to App.css and link to index.css

REFACTOR TO USE LESS

ckopecky committed:
Refactor to use LESS

ckopecky committed:
Refactor to use LESS

ckopecky committed:
Add classname to outer wrapper

ckopecky committed:
Change path to css file

ckopecky committed:
Change minor styling in response to default reset

ckopecky committed:
Format Document

ckopecky committed:
CSS Rewritten after preprocess in LESS

ckopecky committed:
Change pathway to index.css

ckopecky committed:
Add LESS files to gitignore

ckopecky committed:
Import Header to use for Loading component

ckopecky committed:
Add connection to deployed site for mongo

ckopecky committed:
add reactstrap and dependencies

ckopecky committed:
add Reactstrap and dependencies

ckopecky committed:
Add loading to state and to render

ckopecky committed:
bugs in passing props fixed

ckopecky committed:
correct typo

ckopecky committed:
Update README.md to list deployment of backend

ckopecky committed:
add toml file for netlify

ckopecky committed:
Merge branch 'update-notes' of https://github.com/ckopecky/front-end-project-week into update-notes

ckopecky committed:
Add frontend deployment

ckopecky committed:
add deploy to axios call

ckopecky committed:
add deploy to axios call

ckopecky committed:
setState to false even on error

ckopecky committed:
Merge pull request #1 from ckopecky/update-notes

Update notes

ckopecky committed:
Basic Auth Schema set up

ckopecky committed:
Netlify Deploy setup from master now

ckopecky committed:
Basic AuthRouter created

ckopecky committed:
Change Secret to env secret and took out race property

ckopecky committed:
Take out one more instance of user.race

ckopecky committed:
Deleted and reinstalled yarn dependencies

ckopecky committed:
sidebar styles added

ckopecky committed:
Updated styles

ckopecky committed:
Change global less style to include App -

Remove container since not part of this project

Recompiled CSS

Add edit-create styling

ckopecky committed:
Label imports to indicate function

ckopecky committed:
Change tablet to 900px

ckopecky committed:
Remove App to move to global

Add hover on buttons

ckopecky committed:
Remove App and move to global

ckopecky committed:
newly compiled css

ckopecky committed:
Add new single-note less file to index

ckopecky committed:
remove 100vmax

ckopecky committed:
Add single note view style

ckopecky committed:
Merge pull request #2 from ckopecky/update-notes

Update notes

ckopecky committed:
add authcontroller to server

ckopecky committed:
add login route to frontend

ckopecky committed:
add classname to header

ckopecky committed:
login component created

ckopecky committed:
compiled css

ckopecky committed:
add login-register less file to index

ckopecky committed:
Add correct path to authschema

ckopecky committed:
add styling for modal responsiveness

ckopecky committed:
add modal button parametric mixin

ckopecky committed:
adjustments for tablet media query

ckopecky committed:
Instantiate login page styling

ckopecky committed:
Add placeholders and already have an account link

ckopecky committed:
compiled CSS

ckopecky committed:
tablet media queries

ckopecky committed:
add to attribute for link

ckopecky committed:
comment out confirmed Password

ckopecky committed:
login register styling

ckopecky committed:
Change to form

adjust submit handler accordingly

Add logic to set token on localStorage if login successful

ckopecky committed:
add class for register header

ckopecky committed:
add restricted middleware to project

ckopecky committed:
breaking change

ckopecky committed:
remove visible auth stuff that throws errs

TODO: implement fully as a stretch.

ckopecky committed:
Merge pull request #3 from ckopecky/update-notes

Auth stuff

ckopecky committed:
update readme

ckopecky committed:
update readme

ckopecky committed:
update packages

ckopecky committed:
change URI for prod db

ckopecky committed:
fixed connection to mongo

ckopecky committed:
move files

ckopecky committed:
move files

ckopecky committed:
es6 import

ckopecky committed:
add setState to error

ckopecky committed:
update imports and packages

ckopecky committed:
change package json to possibly deploy

ckopecky committed:
change extension on netlify

ckopecky committed:
error handling in front end

ckopecky committed:
no notes found

ckopecky committed:
update mongo string

ckopecky committed:
add faker endpoint
